### ISSUE TITLE
Missing values not handled with spline predictors

### ISSUE THREAD

**Issue by olafdimigen** [unfoldtoolbox/unfold#46]
There appears to be an issue with missing predictor information in combination with spline predictors. 

If a continuous linear predictor contain missing information (NaN or []) in EEG.event.predictorvalues for some events, the corresponding rows of EEG.unfold.X will contain NaN as expected. These are then detected and imputed by uf_imputeMissing(). All fine.

In contrast, if a continuous predictor modeled with splines contain missing informatin (NaN or []), this has not effect on EEG.unfold.X, which will contain valid numbers (rather than NaNs) in all columns of the affected rows of the DM. Consequently, calling uf_imputeMissing() afterwards does nothing and gives feedback that there were no NaNs found in DM (since there aren't any).

TL;DR: Missing values apparently not considered for spline predictors.

**Comment by morenococo** [comment]
yes, and this behaviour was observed also by me when running the toolbox on
a Mac with Matlab R2016b. may be also related to the freezing, although,
not sure about that .

**Comment by behinger** [comment]
Not sure about moreno's point (Could not reproduce so far, maybe you can run the unittests? (goto ```unfold/unittests```  and run ```uf_tests```)).

I fixed this bug, I made a simple mistake on putting nan values from splines to 0 in the designmatrix instead of nan.

https://github.com/unfoldtoolbox/unfold/blob/master/src/uf_toolbox/uf_designmat.m#L567

**Comment by behinger** [comment]
I made a mistake which is now fixed. Now NANs should be propagated properly! Sorry!

**Comment by morenococo** [comment]
Hi guys,

not sure whether you have resolved this issue for other versions of MAC and
Matlab (R 2015a), but it is not yet fully working for me.

I have run this subject fine when I was in Edinburgh using a PC, and a more
recent version of Matlab (R 2016b -- I think), but it is not running now.
I get this error:

uf_glmfit(): Fitting deconvolution model...Matrix is too large to convert
to linear index.

Error in uf_glmfit (line 93)
assert(~any(isnan(EEG.unfold.Xdc(:))),'Warning NAN values
found in designmatrix. will not continue')

I am not even fitting a spline predictor, just a linear one, and I am using
the lastet version of your software. I am a bit stuck as I have no access
to any PCs till September ... could you please advice? I can share this
subject with you (Olaf should already have it --- 14.mat --- from my
study).

thank you guys for developing this little jewel :)

Moreno

**Comment by behinger** [comment]
Hi!
As a hotfix you can simple comment out this line. I most likely have to optimize this assertion

Best,
Bene

Am 13. Juli 2018 18:20:43 MESZ schrieb moreno <notifications@github.com>:

**Comment by morenococo** [comment]
which line? 93?
thx

**Comment by morenococo** [comment]
gotcha, commenting that line works!! not sure what are the implications but
I guess nothing serious.
